sorters/sorter.cpp

```diff
@@ -113,15 +113,12 @@ int Sorter::compareRows(const QModelIndex &source_left, const QModelIndex &sourc
 
 int Sorter::compare(const QModelIndex &sourceLeft, const QModelIndex &sourceRight, const QQmlSortFilterProxyModel& proxyModel) const
 {
-    if (lessThan(sourceLeft, sourceRight, proxyModel))
-        return -1;
-    if (lessThan(sourceRight, sourceLeft, proxyModel))
-        return 1;
-    return 0;
+    return lessThan(sourceLeft, sourceRight, proxyModel) ? -1 : 1;
```

Comment:
To preserve stable sorting we would have to remove `lessThen` function completely and force reimplementing `compare` in subclasses. And e.g. in `ExpressionSorter` should return -1,0,1 then. Otherwise we loose stability of sorting what is not desired imo.

Reply:
`lessThan` doesn't need equal operator. Just what < would return.

Reply:
Ahh, the `compare` / `compareRows` functions are a bit misleading and unnecesary here bc `QQmlSortFilterProxyModel::lessThan` flattens the output to just `bool`. Right? So we are reducing from even 4 comparisons to max 2?

Reply:
I also wonder it it will work correctly with sort inversion via setting different sort order :thinking: